test/integration/json-serialize-original-error/pages/bigint.js

```diff
@@ -0,0 +1,16 @@
+export async function getStaticProps() {
+  return {
+    props: {
+      topics: [
+        {
+          number: '22',
+        },
+      ],
+    },
+  }
+}
+
+export default function Repro(props) {
+  props.topics[0].number = 22n // basically what happened in https://github.com/blitz-js/babel-plugin-superjson-next/issues/63
```

Comment:
AFAICT, this error really only occurs when props are mutated during rendering. Normal JSON-misuse (e.g. bigints) is catched before by custom Next.js logic.
Mutating during render is a big no-no, so errors that aren't about "circular structure" could even be used to indicate the mistake and point towards the React docs.